core/domain/classroom_config_domain.py

```diff
@@ -152,46 +281,102 @@ def require_valid_url_fragment(cls, url_fragment: str) -> None:
 
         if url_fragment == '':
             raise utils.ValidationError(
-                'Url fragment field should not be empty')
+                'Url fragment field should not be empty'
+            )
 
         utils.require_valid_url_fragment(
             url_fragment, 'Classroom URL Fragment',
-            constants.MAX_CHARS_IN_CLASSROOM_URL_FRAGMENT)
+            constants.MAX_CHARS_IN_CLASSROOM_URL_FRAGMENT
+        )
 
-    def validate(self) -> None:
-        """Validates various properties of the Classroom."""
+    @classmethod
+    def require_valid_thumbnail(
```

Comment:
Wait, won't it fail with the existing classroom?

Reply:
No it won't fail as when updating the classroom we will sending all the data and the classroom domain object will have all the fields so it should work just fine.